pkg/reconciler/common/deployments_override_test.go

```diff
@@ -118,6 +119,48 @@ func TestDeploymentsTransform(t *testing.T) {
 			expTemplateAnnotations: map[string]string{"cluster-autoscaler.kubernetes.io/safe-to-evict": "true"},
 			expReplicas:            10,
 		}},
+	}, {
+		name: "override env vars",
```

Comment:
I can add more tests here eg. empty case.

Reply:
@skonto Try to add test cases with more coverage. Otherwise it looks good to me. Wait for @nak3 to see if he has feedback.